code/modules/surgery/organs/wings.dm

```diff
@@ -145,9 +145,57 @@
 	name = "pair of bee wings"
 	desc = "A pair of bee wings. They seem tiny and undergrown"
 	icon_state = "beewings"
-	flight_level = WINGS_FLIGHTLESS
+	flight_level = WINGS_COSMETIC
+	actions_types = list(/datum/action/item_action/organ_action/use/bee_dash)
 	wing_type = "Bee"
 
+/datum/action/item_action/organ_action/use/bee_dash
+	var/jumpdistance = 3
+	var/jumpspeed = 1
+	var/recharging_rate = 100
+	var/recharging_time = 0
+
+/datum/action/item_action/organ_action/use/bee_dash/Trigger()
+	var/mob/living/carbon/L = owner
+
+	if(L.stat != CONSCIOUS || L.buckling) // Has to be concious and unbuckled
+		return
+	if(recharging_time > world.time)
+		to_chat(L, "<span class='warning'>The wings aren't ready to dash yet!</span>")
+		return
+	var/datum/gas_mixture/environment = L.loc.return_air()
+	if(environment && !(environment.return_pressure() > 30))
+		to_chat(L, "<span class='warning'>The atmosphere is too thin for you to dash!</span>")
+		return
+
+	var/turf/target = get_edge_target_turf(L, L.dir) //represents the user's direction
+	var/hoppingtable = FALSE // Triggers the trip
+	var/jumpdistancemoved = jumpdistance // temp jumpdistance
+	var/turf/checkjump = get_turf(L)
+
+	for(var/i in 1 to jumpdistance) //This is how hiero club find the tiles in front of it, tell me/fix it if there's a better way
+		var/turf/T = get_step(checkjump, L.dir)
+		if(locate(/obj/structure/table) in T) // If there's a table, trip
+			hoppingtable = TRUE
+			umpdistancemoved = i
+			break
+		if(!checkjump.Adjacent(T)) // Check for things other than tables that would block flight
```

Comment:
```suggestion
		if(T.density)
			break
		if(!checkjump.Adjacent(T)) // Check for things other than tables that would block flight
```

It seems that `Adjacent` doesn't check turfs' densities because BYOND movement code considers them already. We have to manually check `T`'s density.